pkg/apply/apply.go

```diff
@@ -74,9 +74,44 @@ func ApplyObject(ctx context.Context, client k8sclient.Client, obj *uns.Unstruct
 			}
 
 			return errors.Wrapf(err, "could not update object %s", objDesc)
-		} else {
-			log.Print("update was successful")
 		}
+		log.Print("update was successful")
+	}
+
+	return nil
+}
+
+// DeleteObject deletes an object in the apiserver
+func DeleteObject(ctx context.Context, client k8sclient.Client, obj *uns.Unstructured) error {
+	name := obj.GetName()
+	namespace := obj.GetNamespace()
+	if name == "" {
```

Comment:
When is this the case? Is it needed? All our objects have a name, don't they?

Reply:
just copy pasted from ApplyObject tbh, dont know by heart atm

Reply:
Then please take the time and understand the code you introduce.

Reply:
i did understand to the extend i felt needed at that point but no problem

if we look on the original ApplyObject which has it
it gets rendered objects, that for example were created from yamls at data folder.
if one of them doesnt have a valid name , it will catch it

now lets say someone does delete, i think its a good sanity check that it will at least have a name, for example if file got corrupted

Reply:
aging 7 days auto close, xD